WcaOnRails/app/models/competition.rb

```diff
@@ -142,10 +142,44 @@ def info_for(user)
     info
   end
 
+  attr_accessor :cloned_from_id
+  def cloned_from
+    Competition.find_by(id: cloned_from_id)
+  end
+
   def build_clone
     Competition.new(attributes.slice(*CLONEABLE_ATTRIBUTES)).tap do |clone|
-      clone.organizers = organizers
-      clone.delegates = delegates
+      clone.cloned_from_id = id
+
+      Competition.reflections.keys.each do |association_name|
+        case association_name
+        when 'registrations', 'results', 'competitors', 'competitor_users', 'delegate_report',
+             'competition_delegates', 'competition_organizers', 'media', 'scrambles'
+          # Should be neither cloned nor copied.
```

Comment:
Actually you're right, what I called copying is implicit cloning. AFAIK `clone.organizers = organizers` creates a new row in the `competition_delegates` table with the same `delegate_id`. So `competition.delegates` are exactly the same whereas `competition.tabs` differ.